src/server/useServerUnistyles.tsx

```diff
@@ -10,33 +10,42 @@ declare global {
     }
 }
 
-export const useServerUnistyles = () => {
-    const isServerInserted = useRef(false)
+export const getServerUnistyles = (includeRNWStyles = true) => {
+    if (!isServer()) throw new Error('Server styles should only be read on the server')
+    // @ts-ignore
+    const rnwStyle: string | null = includeRNWStyles ? (StyleSheet?.getSheet().textContent ?? '') : null
+    const css = UnistylesWeb.registry.css.getStyles()
+    const state = UnistylesWeb.registry.css.getState()
+    return <>
+        {rnwStyle && <style id='rnw-style'>{rnwStyle}</style>}
+        <style id='unistyles-web'>{css}</style>
+        <script id='unistyles-script'>{`window.__UNISTYLES_STATE__ = ${JSON.stringify(state)}`}</script>
+    </>
+}
 
-    useServerInsertedHTML(() => {
-        if (!isServerInserted.current) {
-            isServerInserted.current = true
+export const resetServerUnistyles = () => {
+    if (!isServer()) throw new Error('Server styles should only be reset on the server')
```

Comment:
let's use our util for errors, and wrap this in a block statement
```suggestion
    import { error } from '../web/utils'

    if (!isServer()) {
        throw error('Server styles should only be reset on the server')
    }
```